recipes/openssh/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+        "with_pam": False,
+        "with_sandbox": "auto"
+    }
+
+    def _patch_sources(self):
+        print("patching for {}-{}-{}".format(self.version, str(self.settings.os).lower(), self.settings.get_safe("os.version")))
+
+        # general patches
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            print(f"patching general files: {p['patch_file']}")
+            patch(self, **p, base_path=self.source_folder)
+        
+        # os specific patches
+        for p in self.conan_data.get("patches", {}).get("{}-{}".format(self.version, str(self.settings.os).lower()), []):
+            print(f"patching os specific files: {p['patch_file']}")
+            patch(self, **p, base_path=self.source_folder)
+
+        # os version specific patches
+        for p in self.conan_data.get("patches", {}).get("{}-{}-{}".format(self.version, str(self.settings.os).lower(), self.settings.get_safe("os.version")), []):
+            print(f"patching os version specific: {p['patch_file']}")
+            patch(self, **p, base_path=self.source_folder)
+
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
+
+    def export_sources(self):
+        copy(self, "patches/*.patch", self.recipe_folder, self.export_sources_folder, keep_path=True)
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("zlib/[>=1.2.12]")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1q")
+        if self.options.with_pam:
+            self.requires("openpam/20190224")
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD", "Neutrino"]:
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported on {self.settings.os}.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        ad = AutotoolsDeps(self)
+        ad.generate()
+
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.extend([
+            "--with-openssl={}".format("yes" if self.options.with_openssl else "no"), 
+            "--with-pam={}".format("yes" if self.options.with_pam else "no"),
+        ])
+
+        if self.options.with_openssl:       
+            openssl = self.dependencies["openssl"]     
+            tc.configure_args.append("--with-ssl-dir={}".format(openssl.package_folder))
+
+        if self.options.with_sandbox != 'auto':
+            tc.configure_args.append("--with-sandbox={}".format(self.options.with_sandbox))
+
+        tc.generate()
+
+    def build(self):
+        self._patch_sources()
+
+        autotools = Autotools(self)
+        env = VirtualRunEnv(self)
```

Comment:
Any explanation as to why this is needed?

Reply:
When build with `*:shared=True`
This error is produced during configure:
```
configure:15472: cc -o conftest  -m64 -O3 -pipe -Wno-error=format-truncation -Wall -Wpointer-arith -Wuninitialized -Wsign-compare -Wformat-security -Wsizeof-pointer-memaccess -Wno-pointer-sign -Wno-unused-result -Wimplicit-fallthrough -Wmisleading-indentation -fno-strict-aliasing -D_FORTIFY_SOURCE=2 -ftrapv -fzero-call-used-regs=all -fno-builtin-memset -fstack-protector-strong -fPIE -I/home/khj/.conan2/p/b/opensb564e5c3c60b5/p/include  -I/home/khj/.conan2/p/b/opensb564e5c3c60b5/p/include -I/home/khj/.conan2/p/zlib5fd5adbde7f5c/p/include -DNDEBUG -D_XOPEN_SOURCE=600 -D_BSD_SOURCE -D_DEFAULT_SOURCE -L/home/khj/.conan2/p/b/opensb564e5c3c60b5/p/lib  -L/home/khj/.conan2/p/b/opensb564e5c3c60b5/p/lib -L/home/khj/.conan2/p/zlib5fd5adbde7f5c/p/lib -m64 -Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack -fstack-protector-strong -pie conftest.c -lcrypto  -lssl -lcrypto -lz -ldl -lpthread -lrt >&5
configure:15472: $? = 0
configure:15472: ./conftest
./conftest: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory
configure:15472: $? = 127
```

The only way I can let it know where openssl .so files are is with a virtual run env.